components/ContentFlow.brs

```diff
@@ -145,62 +152,186 @@ sub onTruexEvent(event as object)
         m.adRenderer.SetFocus(false)
         m.videoPlayer.control = "play"
     else if data.type = "adError" then
-        ' there was a problem loading the True[X] ad, resume playback with default video ads
+        ' this event is triggered whenever TruexAdRenderer encounters an error
+        ' usually this means the video stream should continue with normal video ads
         m.videoPlayer.control = "play"
     else if data.type = "noAdsAvailable" then
-        ' there are no True[X] ads available for the user to engage with, resume playback with default video ads
+        ' this event is triggered when TruexAdRenderer receives no usable true[X] ad in the ad fetch response
+        ' usually this means the video stream should continue with normal video ads
         m.videoPlayer.control = "play"
     else if data.type = "cancelStream" then
+        ' this event is triggered when the user performs an action interpreted as a request to end the video playback
+        ' this event can be disabled by adding supportsUserCancelStream=false to the TruexAdRenderer init payload
+        ' there are two circumstances where this occurs:
+        '   1. The user was presented with a Choice Card and presses Back
+        '   2. The user has earned an adFreePod and presses Back to exit engagement instead of Watch Your Show button
+
         '
         ' [8]
         '
+
+        ? "TRUE[X] >>> ContentFlow::onTruexEvent() - user requested video stream playback cancel..."
         m.top.event = { trigger: "cancelStream" }
     end if
 end sub
 
-sub onPayload()
-    decodedData = m.sdkLoadTask.payload
+'--------------------------------------------------------------------------------------------------------
+' Callback triggered when ImaSdbTask updates its 'truexAdData' field. The following fields are expected:
+'   * currentAdBreak as roAssociativeArray - current ad break information, including duration
+'   * user_id as string - the identifier used as the network_user_id parameter in the VAST config URL
+'   * vast_config_url as string - the base URL used to request the VAST config
+'   * placement_hash as string - the ad placement's hash value
+'
+' Params:
+'   * event as roAssociativeArray - contains the updated value of ImaSdkTask.truexAdData
+'--------------------------------------------------------------------------------------------------------
+sub onTruexAdDataReceived(event as object)
+    ? "TRUE[X] >>> ContentFlow::onTruexAdDataReceived()"
+
+    decodedData = event.getData()
     if decodedData = invalid then return
+
+    '
+    ' [4]
+    '
+    ? "TRUE[X] >>> ContentFlow::onTruexAdDataReceived() - seeking video position past placeholder ad and pausing..."
+
+    ' pause the stream, which is currently playing a video ad
+    m.videoPlayer.control = "pause"
```

Comment:
should make sure it makes sense to pause at this point -- we might have to pause earlier: if an ad break is supposed to occur at a certain point, we wouldn't want to play through that point in the stream before inserting the ad